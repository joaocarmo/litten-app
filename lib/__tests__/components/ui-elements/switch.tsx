import TestRenderer from 'react-test-renderer'
import ThemeProvider from '@components/theme/provider'
import { UISwitch } from '@ui-elements'

describe('Snapshot test for the "UISwitch" component', () => {
  it('renders correctly without description', () => {
    const element = TestRenderer.create(
      <ThemeProvider>
        <UISwitch label="Some text here" value={true} />
      </ThemeProvider>,
    ).toJSON()
    expect(element).toMatchSnapshot()
  })
  it('renders correctly with description', () => {
    const element = TestRenderer.create(
      <ThemeProvider>
        <UISwitch
          description="Some more text here"
          label="Some text here"
          value={true}
        />
      </ThemeProvider>,
    ).toJSON()
    expect(element).toMatchSnapshot()
  })
})
