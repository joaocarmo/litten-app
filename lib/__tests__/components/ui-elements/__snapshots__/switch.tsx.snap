// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for the "UISwitch" component renders correctly with description 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#c4c4c4",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 12,
        "paddingBottom": 12,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 12,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 15.75,
          "fontWeight": "400",
        }
      }
    >
      Some text here
    </Text>
    <AndroidSwitch
      accessibilityRole="switch"
      enabled={true}
      on={true}
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      thumbTintColor="#c4c4c4"
      trackColorForFalse="#dcdee1"
      trackColorForTrue="#7239f2"
      trackTintColor="#7239f2"
    />
  </View>,
  <Text
    style={
      Array [
        Array [
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "200",
          },
          Object {
            "marginTop": 8,
          },
        ],
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Some more text here
  </Text>,
]
`;

exports[`Snapshot test for the "UISwitch" component renders correctly without description 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#c4c4c4",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 12,
      "paddingBottom": 12,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 12,
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#000000",
        "flex": 1,
        "fontSize": 15.75,
        "fontWeight": "400",
      }
    }
  >
    Some text here
  </Text>
  <AndroidSwitch
    accessibilityRole="switch"
    enabled={true}
    on={true}
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    thumbTintColor="#c4c4c4"
    trackColorForFalse="#dcdee1"
    trackColorForTrue="#7239f2"
    trackTintColor="#7239f2"
  />
</View>
`;
